velox/functions/sparksql/ToJson.h

```diff
@@ -0,0 +1,386 @@
+  } else {
+    p = folly::detail::unsafeTelescope128(buffer, buffer + maxSize, value);
+  }
+  result.append(buffer, p);
+}
+
+template <>
+void toJson<TypeKind::REAL>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<float>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::DOUBLE>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<double>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::VARCHAR>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<Varchar>();
+  folly::json::escapeString(value, result, {});
+}
+
+template <>
+void toJson<TypeKind::TIMESTAMP>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<Timestamp>();
+  // Spark converts Timestamp in ISO8601 format by default.
+  static const auto formatter =
+      functions::buildJodaDateTimeFormatter("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
+          .value();
+  const auto maxSize = formatter->maxResultSize(options.timeZone);
+  char buffer[maxSize];
+  auto size =
+      formatter->format(value, options.timeZone, maxSize, buffer, false, "Z");
+  result.append("\"").append(buffer, size).append("\"");
+}
+
+// Convert complex-type input to Json string.
+template <>
+void toJson<TypeKind::ROW>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto rowType = std::static_pointer_cast<const RowType>(input.type());
+  auto row = input.castTo<DynamicRow>();
+  result.append("{");
+  bool commaBefore = false;
+  for (int i = 0; i < rowType->size(); i++) {
+    auto data = row.at(i);
+    if (data.has_value()) {
+      if (commaBefore) {
+        result.append(",");
+      } else {
+        commaBefore = true;
+      }
+      result.append("\"");
+      result.append(rowType->nameOf(i));
+      result.append("\":");
+      VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
+          toJson, data->kind(), data.value(), result, options);
+    } else if (!options.ignoreNullFields) {
+      if (commaBefore) {
+        result.append(",");
+      } else {
+        commaBefore = true;
+      }
+      result.append("\"");
+      result.append(rowType->nameOf(i));
+      result.append("\":null");
+    }
+  }
+  result.append("}");
+}
+
+template <>
+void toJson<TypeKind::ARRAY>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto arrayView = input.castTo<Array<Any>>();
+  result.append("[");
+  for (int i = 0; i < arrayView.size(); i++) {
+    if (i > 0) {
+      result.append(",");
+    }
+    if (arrayView[i].has_value()) {
+      auto element = arrayView[i].value();
+      VELOX_DYNAMIC_TYPE_DISPATCH_ALL(
```

Comment:
nit: Can we move the dispatch outside the for loop?

Reply:
Each element needs to go through to_json, I'm not quiet sure how to move it outside.